walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.stream.Collectors;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
+import java.util.Set;
 
+/** 
+Algorithm:
+    1. Pick time ranges from events where attendees overlap with meeting request.
+    2. Now find the non overlapping time ranges which have duration more than request duration.
+*/
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    
+    private static final int END_OF_DAY = 24 * 60; 
+    
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {        
+        List<String> requestedAttendees = request.getAttendees().stream()
+                                                 .collect(Collectors.toList());
+        
+        ArrayList<TimeRange> overlappingRanges = calculateOverlappingRanges(events, requestedAttendees);
```

Comment:
List<TimeRange> overlappingRanges like you've declated the List of string abouve

Reply:
Done.